index.js

```diff
@@ -10,6 +10,25 @@ const {
   READ_INDEX
 } = require('./lib/indexes')
 
+const WeakRef = global.WeakRef || class FakeWeakRef {
+  constructor (value) {
+    this._value = value
+  }
+
+  deref () {
+    return this._value
+  }
+}
+
+const FinalizationRegistry = global.FinalizationRegistry || class FakeFinalizationRegistry {
+  register () {}
+  unregister () {}
+}
+
+const registry = new FinalizationRegistry((worker) => {
```

Comment:
Does the `FakeFinalizationRegistry` need a constructor to handle this passed in anonymous function?

Reply:
nope he doesn't. It's just there so I can avoid a few ifs.